.github/workflows/update_readme.yml

```diff
@@ -0,0 +1,36 @@
+# -----------------------------------------------------------------------------
+#   - invoked on PR to main branch
+# -----------------------------------------------------------------------------
+name: update README
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  update-readme:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.11
+      - name: Update README
+        run: |
+          pip install packaging
+          python update_readme.py
+      - name: Commit and push changes
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "github-actions"
+          git diff --quiet && git diff --staged --quiet || (
+            git add README.rst
+            git commit -am "update README with supported Python versions"
```

Comment:
As a side-note, we had discussed PRs automatically committing changes to the changelog, but backed off due to complications about making github perform commits itself. If this works, doesn't that mean we should be able to apply it to that?

Reply:
As discussed, I don't see why this wouldn't work on modifying the changlog